pkg/configmap/store.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package configmap
+
+import (
+	"github.com/knative/pkg/configmap"
+	v1 "k8s.io/api/core/v1"
+)
+
+// TODO Move this to knative/pkg.
+
+// DefaultConstructors is a map for specifying default ConfigMaps to their function constructors.
+//
+// The values of this map must be functions with the definition:
+//
+// func(*k8s.io/api/core/v1.ConfigMap) (... , error)
+//
+// These functions can return any type along with an error.
+type DefaultConstructors map[*v1.ConfigMap]interface{}
```

Comment:
Why is this a map? What are the semantics of a map keyed by struct pointers?

I think this would make more sense to me as:

```go
type DefaultConstructorFunc func() *v1.ConfigMap
type DefaultConstructors map[string]DefaultConstructorFunc
```

I think that would simplify this implementation since there's no need to store default configmaps.

Reply:
I still don't understand why this needs to be a map with a pointer key. This would be equivalent, right?

```go
type DefaultConstructor struct {
  ConfigMap *v1.ConfigMap
  Constructor func(*v1.ConfigMap) (interface{}, error)
}

func NewDefaultUntypedStore(..., defaultConstructors []DefaultConstructor, ...)
```

I still like my version above better, but the array of structs makes more sense to me than the map.

WDYT @Harwayne? Do you _like_ pkg's `configmap.Constructor`?

Reply:
Switched to slice of structs. 

The reason I was using a map to begin with was to protect against the same thing being registered twice. But that protection didn't actually occur. Both because it was a pointer and because I really wanted the key to just be the name of the ConfigMap, not its whole value.